src/main/java/com/inventage/nexusaptplugin/DebianIndexCreator.java

```diff
@@ -279,27 +318,63 @@ public void updateDocument(ArtifactInfo ai, Document doc) {
         }
     }
 
-    private void updateOneDocumentField(ArtifactInfo ai, Document doc, IndexerField indexerField) {
-        if (ai.getAttributes().get(indexerField.getOntology().getFieldName()) != null) {
-            doc.add(indexerField.toField(ai.getAttributes().get(indexerField.getOntology().getFieldName())));
+    /**
+
+     * Updates field **indexerField** of document **doc** for artifact
+     * info **ai**
+
+     *
+     * @param  ai           a map of artifact info
+     * @param  doc          the document to update
+     * @param  indexerField the field to update
+     * @return     nothing again it seems
+     */
+    private void updateOneDocumentField(ArtifactInfo ai,
+                                        Document doc,
+                                        IndexerField indexerField) {
+        String fieldName = indexerField.getOntology().getFieldName();
+        String fieldVal  = ai.getAttributes().get(fieldName);
+        if ( fieldVal == null ){
+          // TODO: throw exception?
+          return;
         }
+        Field field = indexerField.toField(fieldVal);
+        doc.add(field);
     }
 
     @Override
     public boolean updateArtifactInfo(Document doc, ArtifactInfo ai) {
         String filename = doc.get(FILENAME.getKey());
-        if (filename != null && filename.endsWith(".deb")) {
-            for (IndexerField indexerField : indexerFields) {
-                updateOneArtifactInfoAttribute(doc, ai, indexerField);
-            }
-            ai.md5 = doc.get(MD5.getKey());
-            return true;
+        if ( filename == null ){
+          // TODO: throw an exception?
+          return false;
         }
-        return false;
+        if ( ! filename.endsWith(".deb")) {
+          // TODO: throw an exception?
+          return false;
+        }
+
+        for (IndexerField indexerField : indexerFields) {
+          try { 
+            updateOneArtifactInfoAttribute(doc, ai, indexerField);
+          }catch(Exception e){
+            // TODO: rethrow?
+            return false;
+          }
+        }
+        ai.md5 = doc.get(MD5.getKey());
+        // TODO: sha1? sha256?  sha384?  sha512?
```

Comment:
The md5 hash is part of the API provided by Nexus. All the other hashes are custom attributes and saved as such. See line 281.